_docs/src/documentation/test_projects.md

```diff
@@ -0,0 +1,348 @@
+# How To Create Test Projects
+
+## Type Of Test Project
+
+Rex supports 3 different kind of test projects, each requiring a slightly different setup.
+
+### Unit Test Project
+A unit test project is ideal to test a library or system that doesn't need to engine to be in a specific state. If you have a container, allocator, algorithm, .. that you'd like to test, a unit test is ideal to do this.
+
+Let's work from an example. Let's pretend we want to create a unit test project for the rex engine project called `RexUnitTests`.
+To create a unit test project, create a new folder in ~/tests under the specific subfolder you need, in this case, because we're making a unit test for the engine itself, we'll make a folder under `1_engine`.
+
+Like any other project, we need to create a `src` and `include` folder and a `sharpmake file`.
+
+this would give us the following folder structure
+- ~/tests/1_engine/rex_unit_tests/include
+- ~/tests/1_engine/rex_unit_tests/src
+- ~/tests/1_engine/rex_unit_tests/rex_unit_tests.sharpmake.cs
+
+if needed, you also need to create a code generation config file under `~/config/1_engine/rex_unit_tests`. For this example, we don't need such a file.
+
+Next you need to configure the sharpmake project. You do this like you'd configure any other project, but every unit test project needs 2 things to work properly.
+- The `ProjectType` needs to be specified in the constructor. Because we're generating a unit test, specify `ProjectType = ProjectGen.TestProjectType.UnitTest;`.
+- We need to add a dependency to `RexUnitTest` project. Override the `SetupLibDependencies` func and add a public dependency to `RexUnitTest`.
+
+An example of such a file could look like this
+
+```cs
+using Sharpmake;
+using System.IO;
+
+[Generate]
+public class RexEngineUnitTests : TestProject
```

Comment:
Not sure if there is any documentation yet about these base projects users can inherit from but that would be ideal. Lot of boiler plate needs to happen as well seems like this could benefit from some automation IMO. I think most unit tests projects would look very similar